com.unity.perception/Runtime/Randomization/Randomizers/OrderedSet.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace UnityEngine.Perception.Randomization.Randomizers
+{
+    /// <summary>
+    /// This collection has the properties of a HashSet that also preserves insertion order. As such, this data
+    /// structure demonstrates the following time complexities:
+    /// O(1) lookup, O(1) insertion, O(1) removal, and O(n) traversal
+    /// </summary>
+    /// <typeparam name="T">The item type to store in this collection</typeparam>
+    class OrderedSet<T> : ICollection<T>
```

Comment:
I think the name "OrderedSet" refers to a different type of data structure. What you have here is similar to Java's LinkedHashSet. We could consider changing the name.

Reply:
Done